### DISCUSSION TITLE
Unable to load picture or PDF file

### DISCUSSION THREAD

**Discussion by rjss424** [quarto-dev/quarto-cli#4397] [Q&A] [answered]
Please let me know if it's been answered elsewhere.
I can see my figures printed beneath my code block, but I get an error when Quarto tries to render to html or pdf. 

Installed Quarto and tinytex:
```
$ quarto list tools; quarto --version
[>] Inspecting tools

Tool         Status            Installed     Latest
chromium     Not installed     ---           869685
tinytex      Up to date        v2023.02      v2023.02
1.2.335

$ quarto --version
1.2.335
```

I get an error saying the pdf (and also html render) was not able to find the image, although the image is printed in the subdir.

The hello.qmd renders just fine and I'm able to view the ggplot() example in both output renders.

This is the error I get:
```
compilation failed- error
Unable to load picture or PDF file '"filename/figure-pdf/error-line-plot-1.pdf"'.
<to be read again> 
```

**Comment by dragonstyle** [comment]
Can you share the source code / directory structure that is causing the issue (or a minimal example set of code we could take a look at)?

**Reply by rjss424** [reply]
Hi, sorry for any inconvenience. I was making sample source code/data to recreate the issue, but happened to fix my own issue in doing so.

Not sure what it was, maybe it was a space in the .qmd filename? 

For more context, here's the printout error from the .log file.
I also attached the sample source code/data I created to hopefully recreate my issue.

```
LaTeX Warning: File `testdata 20230216_files/figure-pdf/error-line-plot-1.pdf' not found on input line 303.

! Unable to load picture or PDF file '"testdata 20230216_files/figure-pdf/error-line-plot-1.pdf"'.
<to be
 read again> 
                   }
l.303 ...6_files/figure-pdf/error-line-plot-1.pdf}
```

[RLocal.zip](https://github.com/quarto-dev/quarto-cli/files/10762052/RLocal.zip)

**Reply by rjss424** [reply]
The issue was not seeing the figure printout in the pdf or html render, and it auto-generated subfolders which stored the printed figures.

**Reply by jamessilva0070** [reply]
Hi.

I overcome this problem by first using ggsave to store PNG output images inside a /images folder, and then using Rmarkdown syntax to insert an image. You can test this approach using the following testing qmd code.

```{r}
#THIS CHUNK DOES NOT RENDER
#library(tidyverse)
#mtcars %>%  
#  ggplot(aes(x = cyl, y = mpg, group = cyl, color = factor(cyl))) +
#  geom_boxplot() +
#  geom_jitter() +
#  ggtitle("A typical example plot with default features")
```

```{r}
# Remember to create the images subfolder in project directory
library(tidyverse)
ggsave("images/mtcarsFig.png",
       mtcars %>%  
  ggplot(aes(x = cyl, y = mpg, group = cyl, color = factor(cyl))) +
  geom_boxplot() +
  geom_jitter() +
  ggtitle("A typical example plot with default features"),
       width=7.5, height=7.5/1.618, units="in"
)
```

### use this line to insert image
![image](https://github.com/quarto-dev/quarto-cli/assets/115560698/d33b0ed2-20a7-485c-9bc3-c1c900c44c95)

**Reply by cderv** [reply, marked as answer]
> `testdata 20230216_files/figure-pdf/error-line-plot-1.pdf' not found on input line 303.

@rjsinahon Did  you solve your issue ? I see a space in the folder (`testdata 20230216_files`). Do you have a space in your Qmd filename ? 

@jamessilva0070 As it seems you have encountered the same issue, do you also have a space in your filename by any chance ? 

LaTeX does not like that if I remember correctly, and we should probably try to take this into account. I want to confirm.

**Reply by rjss424** [reply]
Yes if I recall, I think this was the issue. And if I remember, I made sure to replace any spaces with underscores. I'll go ahead and close this thread now.
